code/game/machinery/cell_charger.dm

```diff
@@ -52,17 +54,19 @@
 			user.visible_message("[user] inserts a cell into [src].", "<span class='notice'>You insert a cell into [src].</span>")
 			chargelevel = -1
 			updateicon()
-	else if(istype(W, /obj/item/wrench))
-		if(charging)
-			to_chat(user, "<span class='warning'>Remove the cell first!</span>")
+	else 
+		if(!charging && default_deconstruction_screwdriver(user, icon_state, icon_state, W))
+			return
+		if(default_deconstruction_crowbar(W))
+			return
+		if(!charging && default_unfasten_wrench(user, W))
 			return
-
-		anchored = !anchored
-		to_chat(user, "<span class='notice'>You [anchored ? "attach" : "detach"] [src] [anchored ? "to" : "from"] the ground</span>")
-		playsound(src.loc, W.usesound, 75, 1)
-	else
 		return ..()
 
+/obj/machinery/cell_charger/Destroy()
+	if(charging)
+		charging.forceMove(drop_location())

```

Comment:
It should just QDEL_NULL it if it's supposed to be inside it

Reply:
Eh if someone beats it till it breaks the battery should fall out of it though? I mean it's ontop of it.

Reply:
I can make it QDELL_NULL but I am not sure it should eat the battery really

Reply:
Deconstruct should return the battery instead if I understand correctly

Reply:
Let me double check real quick

Reply:
deconstuct can move it, Destroy should delete it